crates/hir_def/src/attr.rs

```diff
@@ -656,8 +648,7 @@ fn get_doc_string_in_attr(it: &ast::Attr) -> Option<ast::String> {
 }
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
-pub(crate) struct AttrId {
-    is_doc_comment: bool,
+pub struct AttrId {
```

Comment:
This was using separate numbering because `MacroCallKind` stores the raw attribute index for attribute and derive macros. I think this change breaks the following case:

```rust
/// doc0
#[proc::noop_attr]
#[should_be_visible]
struct S;
```

`noop_attr` previously saw `#[should_be_visible]`, but it now gets removed because `#[proc::noop_attr]` has `ast_index` 1, but it's at index 0 when using `HasAttrs::attrs().skip(ast_index)`, and `#[should_be_visible]` has index 1 in the AST.

Reply:
Ah so it had an extra usage, aight I'll check that later